code/modules/clothing/head/_head.dm

```diff
@@ -15,6 +19,32 @@
 		var/mob/living/carbon/human/H = loc
 		H.update_hair()
 
+///Special throw_impact for hats to frisbee hats at people to place them on their heads/attempt to de-hat them.
+/obj/item/clothing/head/throw_impact(atom/hit_atom, datum/thrownthing/thrownthing)
+	. = ..()
+	///if the thrown object exists, has a target_zone from whatever threw it, isn't the thrower and the target zone isn't the head	
+	if(thrownthing && thrownthing.target_zone && thrownthing.thrower != hit_atom && thrownthing.target_zone != BODY_ZONE_HEAD)
+		return
+	if(iscarbon(hit_atom))
+		var/mob/living/carbon/H = hit_atom
+		if(istype(H.head, /obj/item/clothing/head))
+			var/obj/item/clothing/head/WH = H.head
+			///if H's head slot has something equipped, has snug_fit set to FALSE, and the thrown hat has anti_tinfoil_maneuver set to FALSE
+			if(istype(WH) && !WH.snug_fit && !anti_tinfoil_maneuver)
+				///attempt to drop the equipped item to the ground and attempt to equip the thrown hat
+				if(H.dropItemToGround(WH) && H.equip_to_slot_if_possible(src, SLOT_HEAD, 0, 1, 1))
+					H.visible_message("<span class='warning'>[src] knocks [WH] off [H]'s head!</span>", "<span class='warning'>[WH] is suddenly knocked off your head, replaced by [src]!</span>")
+			else
+				H.visible_message("<span class='warning'>[src] bounces off [H]'s [WH.name]!", "<span class='warning'>[src] bounces off your [WH.name], falling to the floor.</span>")
+				return
+		///H has nothing equipped in the head slot, so the thrown hat will attempt to equip itself if anti_tinfoil_maneuver is set to FALSE		
+		if(!H.head && !anti_tinfoil_maneuver && H.equip_to_slot_if_possible(src, SLOT_HEAD, 0, 1, 1))

```

Comment:
You should just abort the entire carbon section if anti_tinfoil_maneuver is true.
Also try to only not have a duplicate of ``H.equip_to_slot_if_possible(src, SLOT_HEAD, 0, 1, 1)``